pkg/api/validation/validation.go

```diff
@@ -2844,6 +2854,8 @@ func ValidateService(service *api.Service) field.ErrorList {
 		allErrs = append(allErrs, field.NotSupported(specPath.Child("sessionAffinity"), service.Spec.SessionAffinity, supportedSessionAffinityType.List()))
 	}
 
+	allErrs = append(allErrs, validateClientIPAffinityConfig(service.Spec.ClientIPAffinityConfig, specPath.Child("clientIPAffinityConfig"))...)
```

Comment:
> You need to check which sessionAffinity setting is in use. If it is None, don't verify.

I check the sessionAffinity setting inside `validateClientIPAffinityConfig()`. PTAL. THANKS!

> and if it is ClientIP, require that the config is non-nil.

If it is ClientIP, I create a default config. Is that ok?

> also we should probably document and enforce an upper bound. 86400 for 1 day? 604800 for 1 week?

Current upper bound is 86400 for 1 day. Is that ok?